public/app/features/explore/TimePicker.test.tsx

```diff
@@ -1,74 +1,238 @@
 import React from 'react';
 import { shallow } from 'enzyme';
 import sinon from 'sinon';
+import moment from 'moment';
 
+import * as dateMath from 'app/core/utils/datemath';
 import * as rangeUtil from 'app/core/utils/rangeutil';
-import TimePicker, { DEFAULT_RANGE, parseTime } from './TimePicker';
+import TimePicker from './TimePicker';
+import { RawTimeRange, TimeRange, TIME_FORMAT } from '@grafana/ui';
+
+const DEFAULT_RANGE = {
+  from: 'now-6h',
+  to: 'now',
+};
+
+const fromRaw = (rawRange: RawTimeRange): TimeRange => {
+  const raw = {
+    from: moment.isMoment(rawRange.from) ? moment(rawRange.from) : rawRange.from,
+    to: moment.isMoment(rawRange.to) ? moment(rawRange.to) : rawRange.to,
+  };
+
+  return {
+    from: dateMath.parse(raw.from, false),
+    to: dateMath.parse(raw.to, true),
+    raw: rawRange,
+  };
+};
 
 describe('<TimePicker />', () => {
-  it('renders closed with default values', () => {
-    const rangeString = rangeUtil.describeTimeRange(DEFAULT_RANGE);
-    const wrapper = shallow(<TimePicker />);
-    expect(wrapper.find('.timepicker-rangestring').text()).toBe(rangeString);
-    expect(wrapper.find('.gf-timepicker-dropdown').exists()).toBe(false);
+  it('render default values when closed and relative time range', () => {
```

Comment:
Not at all satisfied with these tests. Recommended to use snapshots instead to verify the rendered markup?

Reply:
Since these are render specific tests I would opt for snapshots. Especially tat you rely on class names which 1) tend to change over time 2) sooner or later will be replaced with Emotion class names, which won't be that testable.